tests/load/prometheus.go

```diff
@@ -1,118 +0,0 @@
-// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
-// See the file LICENSE for licensing terms.
-
-package load
-
-import (
-	"context"
-	"encoding/json"
-	"errors"
-	"fmt"
-	"net"
-	"net/http"
-	"os"
-	"path/filepath"
-	"time"
-
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
-	"go.uber.org/zap"
-
-	"github.com/ava-labs/avalanchego/tests/fixture/tmpnet"
-	"github.com/ava-labs/avalanchego/utils/logging"
-	"github.com/ava-labs/avalanchego/utils/perms"
-)
-
-type MetricsServer struct {
-	addr     string
-	registry *prometheus.Registry
-	server   http.Server
-}
-
-func NewPrometheusServer(addr string, registry *prometheus.Registry) *MetricsServer {
-	return &MetricsServer{
-		addr:     addr,
-		registry: registry,
-	}
-}
-
-func (*MetricsServer) String() string {
-	return "metrics server"
-}
-
-func (s *MetricsServer) Start() (runError <-chan error, err error) {
-	const metricsPattern = "/ext/metrics"
-
-	mux := http.NewServeMux()
-	handlerOpts := promhttp.HandlerOpts{Registry: s.registry}
-	mux.Handle(metricsPattern, promhttp.HandlerFor(s.registry, handlerOpts))
-
-	listener, err := net.Listen("tcp", s.addr)
-	if err != nil {
-		return nil, err
-	}
-	s.addr = listener.Addr().String()
-
-	s.server = http.Server{
-		Addr:              s.addr,
-		Handler:           mux,
-		ReadHeaderTimeout: time.Second,
-		ReadTimeout:       time.Second,
-	}
-
-	runErrorBiDirectional := make(chan error)
-	runError = runErrorBiDirectional
-	ready := make(chan struct{})
-	go func() {
-		close(ready)
-		err = s.server.Serve(listener)
-		if errors.Is(err, http.ErrServerClosed) {
-			return
-		}
-		runErrorBiDirectional <- err
-	}()
-	<-ready
```

Comment:
I was not sure the purpose of this, but have not seen any related issues after removing, so simplified this and moved it to another file where it can be shared.